sigmf-spec.md

```diff
@@ -200,10 +200,15 @@ Rules for SigMF Collection files:
 Rules for SigMF Archive files:
 1. The Archive MUST use the `tar` archive format, as specified by POSIX.1-2001.
 1. The Archive file's filename extension MUST be `.sigmf`.
-1. The Archive MUST contain the following files: for each contained Recording
-   with some name given here meta-syntactically as `N`, files named `N` (a
-   directory), `N/N.sigmf-meta`, and `N/N.sigmf-data`.
-1. The Archive MAY contain a `.sigmf-collection` file in the top-level directory.
+1. The Archive MUST contain at least one SigMF Recording.
+1. The Archive MAY organize SigMF Recordings into subdirectories such that for
+   each contained Recording with some base name `N`, individual files are stored
+   within the directory `N` (e.g.: `N/N.sigmf-meta`, and `N/N.sigmf-data`). The
```

Comment:
I suggest that all references to `N/N` be removed, and just say that any directory structure can be used -- the only requirement is that the `.sigmf-meta` file be in the same directory (and have the same name stem) as the `.sigmf-data` file.
For Non-Conforming Datasets, I don't see a good reason for there to be any correlation between the directory/name of the `.sigmf-meta` file and the `dataset` field.

Alternative language:
"The Archive MAY organize SigMF Recordings into subdirectories, but the `.sigmf-meta` and the `.sigmf-data` files must be in the same directory (except for Non-Conforming Datasets, see the description of the `dataset` field below)."

Reply:
Permitting arbitrary directory structures places a somewhat obnoxious burden on parsing applications to do directory traversal through the archive, and it obfuscates how many Recordings are present in an archive. The benefit of permitting this seems negligible at best. Permitting more than one Recording in an Archive is a bit OBE now that SigMF has Collections anyway.

Reply:
If the user is accessing the data without extracting the contents of the archive, then I see no difference in usage -- that is to say, I don't see anything obnoxious about that. If the user *wants* to double the disk usage by extracting an archive (yeah, beats me why you would want to do that <wink>) then the `--strip-components` switch to `tar` can eliminate the directory structure entirely and just leave you with files at the base of the tree.

That being said, I'm not worried about it if I don't prevail in this -- there are more important issues than this. I was trying to gently push back on the (IMHO unnecessary) extra stipulation that recording files *should* go in the base of the tar tree -- in the interest of a shorter spec.

Reply:
We definitely want to support accessing data without extracting archive contents, your examples there are excellent.

I'll leave this open for a bit and get more input. I wont unilaterally force a solution either way. I just am sensitive to anything that makes sigmf application development more difficult.